lightning/src/routing/gossip.rs

```diff
@@ -266,9 +282,16 @@ impl<L: Deref> EventHandler for NetworkGraph<L> where L::Target: Logger {
 						self.channel_failed(short_channel_id, is_permanent);
 					},
 					NetworkUpdate::NodeFailure { ref node_id, is_permanent } => {
-						let action = if is_permanent { "Removing" } else { "Disabling" };
-						log_debug!(self.logger, "{} node graph entry for {} due to a payment failure.", action, node_id);
-						self.node_failed(node_id, is_permanent);
+						if is_permanent {
+							#[cfg(feature = "std")]
+							log_debug!(self.logger,
+								"Removed node graph entry for {} due to a payment failure. (Tracking)", node_id);
+							#[cfg(not(feature = "std"))]
+							log_debug!(self.logger,
+								"Removed node graph entry for {} due to a payment failure. (Not tracking)", node_id);
```

Comment:
Initially, we didn't remove it from the graph for `no-std` but thought we should do it anyway, even if we're not tracking. I agree it looks a little pointless to have two messages now since they're effectively the same. Will simplify to one :+1: